src/tqec/_cli/subcommands/dae2circuits.py

```diff
@@ -0,0 +1,105 @@
+            "--out-dir",
+            help="Directory to save the generated stim circuits.",
+            type=Path,
+            required=True,
+        )
+        parser.add_argument(
+            "-k",
+            help="The scale factors applied to the circuits.",
+            nargs="+",
+            type=int,
+            required=True,
+        )
+        parser.add_argument(
+            "--obs-include",
+            help="The observable indices to be included in the circuits. If not provided, all potential observables will be included.",
+            nargs="*",
+            type=int,
+        )
+        parser.add_argument(
+            "--add-detectors",
+            help="Whether to add detectors to the circuits.",
+            action="store_true",
+        )
+        # TODO: add noise models
+        parser.set_defaults(func=Dae2CircuitsTQECSubCommand.execute)
+
+    @staticmethod
+    @override
+    def execute(args: argparse.Namespace) -> None:
+        dae_absolute_path: Path = args.dae_file.resolve()
+        out_dir: Path = args.out_dir.resolve()
+        if not args.out_dir.exists():
+            out_dir.mkdir(parents=True)
+
+        # Construct the block graph from the given .dae file
+        block_graph = BlockGraph.from_dae_file(
+            dae_absolute_path, graph_name=str(dae_absolute_path.name)
+        )
+
+        # Convert to ZX graph and find observables
+        zx_graph = block_graph.to_zx_graph()
+        correlation_surfaces = zx_graph.find_correlation_subgraphs()
+        obs_indices: list[int] = args.obs_include
+        if not obs_indices:
+            obs_indices = list(range(len(correlation_surfaces)))
+        if max(obs_indices) >= len(correlation_surfaces):
+            raise ValueError(
+                f"Found {len(correlation_surfaces)} observables, but requested indices up to {max(obs_indices)}."
+            )
+
+        # Save the plotted observables to a subdirectory
+        observable_out_dir = out_dir / "observables"
+        observable_out_dir.mkdir(exist_ok=True)
+        save_correlation_surfaces_to(
+            zx_graph,
+            observable_out_dir,
+            [correlation_surfaces[i] for i in obs_indices],
+        )
+        # TODO: avoid recomputing the observables
+        abstract_observables = block_graph.get_abstract_observables()
+
+        # Compile the block graph and generate stim circuits
+        circuits_out_dir = out_dir / "circuits"
+        circuits_out_dir.mkdir(exist_ok=True)
+        compiled_graph = compile_block_graph(
+            block_graph,
+            observables=[abstract_observables[i] for i in obs_indices],
+        )
+        ks: list[int] = args.k
+        add_detectors: bool = args.add_detectors
+        for k in ks:
+            circuit = compiled_graph.generate_stim_circuit(k)
+            if add_detectors:
+                circuit = annotate_detectors_automatically(circuit)
+            circuit.to_file(circuits_out_dir / f"{k=}.stim")
```

Comment:
Actually the file name with be exapanded to `k=?` when using `f"{k=}"` fstring. This is [a feature added in python3.8](https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging). And I used `=` instead of `_` because `sinter` expect `key=value` format in circuit name to automatically parse the key-value pairs into `sinter.TaskStats.json_metadata`.